lms/static/js/instructor_dashboard/data_download.js

```diff
@@ -0,0 +1,399 @@
+                dataDownloadObj.$download_display_table.text(gettext('Loading'));
+                return $.ajax({
+                    type: 'POST',
+                    dataType: 'json',
+                    url: url,
+                    error: function() {
+                        dataDownloadObj.clear_display();
+                        return dataDownloadObj.$download_request_response_error.text(
+                            gettext('Error getting student list.')
+                        );
+                    },
+                    success: function(data) {
+                        var $tablePlaceholder, columns, feature, grid, gridData, options;
+                        dataDownloadObj.clear_display();
+                        options = {
+                            enableCellNavigation: true,
+                            enableColumnReorder: false,
+                            forceFitColumns: true,
+                            rowHeight: 35
+                        };
+                        columns = (function() {
+                            var i, len, ref, results;
+                            ref = data.queried_features;
+                            results = [];
+                            for (i = 0, len = ref.length; i < len; i++) {
+                                feature = ref[i];
+                                results.push({
+                                    id: feature,
+                                    field: feature,
+                                    name: data.feature_names[feature]
+                                });
+                            }
+                            return results;
+                        }());
+                        gridData = data.students;
+                        $tablePlaceholder = $('<div/>', {
+                            class: 'slickgrid'
+                        });
+                        dataDownloadObj.$download_display_table.append($tablePlaceholder);
+                        grid = new window.Slick.Grid($tablePlaceholder, gridData, columns, options);
+                        return grid;
+                    }
+                });
+            });
+            this.$list_problem_responses_csv_btn.click(function() {
+                var url;
+                dataDownloadObj.clear_display();
+                url = dataDownloadObj.$list_problem_responses_csv_btn.data('endpoint');
+                return $.ajax({
+                    type: 'POST',
+                    dataType: 'json',
+                    url: url,
+                    data: {
+                        problem_location: dataDownloadObj.$list_problem_responses_csv_input.val()
+                    },
+                    error: function(error) {
+                        dataDownloadObj.$reports_request_response_error.text(
+                            JSON.parse(error.responseText)
+                        );
+                        return $('.msg-error').css({
```

Comment:
I'll stop commenting on lines that don't need returns, you get the point 😛 

If you could, just go through and do a pass for returns that probably aren't needed?